packages/compass-collection/src/stores/context.tsx

```diff
@@ -257,6 +262,71 @@ const setupScopedModals = ({
   return [];
 };
 
+/**
+ * Setup the query bar plugins. Need to instantiate the store and actions
+ * and put them in the app registry for use by all the plugins. This way
+ * there is only 1 query bar store per collection tab instead of one per
+ * plugin that uses it.
+ */
+const setupQueryPlugins = ({
+  globalAppRegistry,
+  localAppRegistry,
+  serverVersion,
+  state,
+  namespace,
+  isReadonly,
+  isTimeSeries,
+  isClustered,
+  isFLE,
+  query,
+  aggregation,
+}: ContextWithAppRegistry): void => {
+  const queryBarRole = globalAppRegistry.getRole('Query.QueryBar')?.[0];
+  if (queryBarRole) {
+    localAppRegistry.registerRole('Query.QueryBar', queryBarRole);
+    const queryBarActions = setupActions(queryBarRole, localAppRegistry);
+    setupStore({
+      role: queryBarRole,
+      globalAppRegistry,
+      localAppRegistry,
+      dataService: state.dataService,
+      namespace,
+      serverVersion,
+      isReadonly,
+      isTimeSeries,
+      isClustered,
+      isFLE,
+      actions: queryBarActions,
+      query,
+      aggregation,
+    });
+  }
+
+  const queryHistoryRole = globalAppRegistry.getRole('Query.QueryHistory')?.[0];
```

Comment:
will this change be required also in DE?

Reply:
No, DE doesn't use query history so it's disabled and this role fetching never happens on their end.  Down the line if we add query history to DE we'll need to register this role and initialize it separately.

Reply:
awesome thanks!